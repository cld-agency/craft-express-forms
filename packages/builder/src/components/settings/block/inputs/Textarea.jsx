import React from 'react';
import BaseInput from './BaseInput';

class Textarea extends BaseInput {
  renderInput() {
    const { name, value, saveHandler } = this.props;

    return <textarea className="text nicetext" name={name} onChange={saveHandler} defaultValue={value} />;
  }
}

export default Textarea;
