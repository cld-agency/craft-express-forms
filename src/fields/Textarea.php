<?php

namespace Solspace\ExpressForms\fields;

use yii\db\Schema;

class Textarea extends BaseField
{
    /**
     * @return string
     */
    public function getType(): string
    {
        return self::TYPE_TEXTAREA;
    }

    /**
     * @inheritdoc
     */
    public function getContentColumnType(): string
    {
        return Schema::TYPE_TEXT;
    }
}
