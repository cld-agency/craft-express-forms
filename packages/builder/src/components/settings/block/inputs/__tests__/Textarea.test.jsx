import { shallow } from 'enzyme';
import React from 'react';
import Textarea from '../Textarea';

describe('<Textarea />', () => {
  let wrapper;
  let onChange;
  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(<Textarea label="A Test Textarea" name="test_textarea" value="a value" saveHandler={onChange} />);
  });

  it('renders & displays correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains a textarea field', () => {
    expect(wrapper.find("textarea[name='test_textarea']")).toHaveLength(1);
  });

  it('changes value', () => {
    const value = 'A different value';
    wrapper.find("textarea[name='test_textarea']").simulate('change', { target: { value } });
    expect(onChange).toBeCalledWith({ target: { value: 'A different value' } });
  });
});
