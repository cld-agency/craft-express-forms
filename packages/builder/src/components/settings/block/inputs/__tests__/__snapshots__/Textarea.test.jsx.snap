// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> renders & displays correctly 1`] = `
<div
  className="block-row"
>
  <div
    className="label"
  >
    <WithStyles(ForwardRef(Tooltip))
      placement="bottom"
      title=""
    >
      <label
        className=""
      >
        A Test Textarea
      </label>
    </WithStyles(ForwardRef(Tooltip))>
  </div>
  <div
    className="input ltr"
  >
    <textarea
      className="text nicetext"
      defaultValue="a value"
      name="test_textarea"
      onChange={[MockFunction]}
    />
  </div>
</div>
`;
